test/e2e/settings/custom_header.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package settings
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/parnurzeal/gorequest"
+
+	"k8s.io/ingress-nginx/test/e2e/framework"
+)
+
+var _ = framework.IngressNginxDescribe("Add custom header", func() {
+	f := framework.NewDefaultFramework("custom-header")
+	host := "custom-header"
+
+	BeforeEach(func() {
+		f.NewEchoDeployment()
+		ing := framework.NewSingleIngress(host, "/", host, f.Namespace, framework.EchoService, 80, nil)
+		f.EnsureIngress(ing)
+	})
+
+	AfterEach(func() {
+	})
+
+	It("Add a custom header", func() {
+		customHeader := "X-A-Custom-Header"
+		customHeaderValue := "customHeaderValue"
+
+		h := map[string]string{}
+		h[customHeader] = customHeaderValue
+
+		f.CreateConfigMap("add-headers-configmap", h)
+
+		wlKey := "add-headers"
+		wlValue := f.Namespace + "/add-headers-configmap"
+
+		f.UpdateNginxConfigMapData(wlKey, wlValue)
+
+		f.WaitForNginxConfiguration(func(server string) bool {
+			return strings.Contains(server, fmt.Sprintf(`more_set_headers "%v": "%v";`, customHeader, customHeaderValue))
```

Comment:
@SzekeresB reading the failure and the dump of the output, https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/ingress-nginx/5065/pull-ingress-nginx-e2e-1-17/1228251500334026752#1:build-log.txt%3A2730  this should be `more_set_headers "%v: %v";`

(check this works running the e2e test locally)